esp/bindings/http/platform/httpbinding.cpp

```diff
@@ -1823,25 +1835,96 @@ int EspHttpBinding::onGetVersion(IEspContext &context, CHttpRequest* request, CH
     StringBuffer srvQName;
     qualifyServiceName(context, service, NULL, srvQName, NULL);
     verxml.appendf("<VersionInfo><Service>%s</Service><Version>%.3f</Version></VersionInfo>", srvQName.str(), m_wsdlVer);
-    
+
     response->setContent(verxml.str());
     response->setContentType(HTTP_TYPE_APPLICATION_XML_UTF8);
     response->setStatus(HTTP_STATUS_OK);
     response->send();
     return 0;
 }
 
+void getServiceSchema(const char *serviceQName, const char *methodQName, double version, const char *nstr,
+                      const StringBuffer &xmlFilename,  Owned<IProperties> &params, EsdlXslTypeId EsdlXslType,
+                      IProperties *opts, StringBuffer &schema)
+{
+    Owned<IEsdlDefinition> esdlDef;
+    Owned<IEsdlDefinitionHelper> defHelper;
+
+    esdlDef.set(createEsdlDefinition(nullptr, nullptr));
+    defHelper.set(createEsdlDefinitionHelper());
+
+    std::string xmlFilePath(getCFD());
+    xmlFilePath.append("esdl_files").append(PATHSEPSTR).append(xmlFilename.str());
+    esdlDef->addDefinitionsFromFile(xmlFilePath.c_str());
```

Comment:
We should keep in mind for the future that we could cache some esdlDefs loaded from the xml files if accessing these is  perceived to be slow.